### PR TITLE
secure rest tests for threat intel monitor apis

### PR DESCRIPTION
**Pull request by eirsep**
### Description
Adds secure rest tests for threat intel monitor APIs
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/test/java/org/opensearch/securityanalytics/resthandler/SecureThreatIntelMonitorRestApiIT.java [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+
+    static String TEST_IT_BACKEND_ROLE = "IT";
+
+    static Map<String, String> roleToPermissionsMap = Map.ofEntries(
+            Map.entry(SECURITY_ANALYTICS_FULL_ACCESS_ROLE, "cluster:admin/opendistro/securityanalytics/detector/*"),
+            Map.entry(SECURITY_ANALYTICS_READ_ACCESS_ROLE, "cluster:admin/opendistro/securityanalytics/detector/read")
+    );
+
+    private RestClient userClient;
+    private final String user = "userDetector";
```
Comment (jowg-amazon) [inline src/test/java/org/opensearch/securityanalytics/resthandler/SecureThreatIntelMonitorRestApiIT.java:55]: nit: rename from detectore

File: src/test/java/org/opensearch/securityanalytics/resthandler/SecureThreatIntelMonitorRestApiIT.java [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+
+    @After
+    public void cleanup() throws IOException {
+        if (userClient != null) {
+            userClient.close();
+        }
+        deleteUser(user);
+    }
+
+    private final String iocIndexMappings = "\"properties\": {\n" +
```
Comment (AWSHurneyt) [inline src/test/java/org/opensearch/securityanalytics/resthandler/SecureThreatIntelMonitorRestApiIT.java:74]: Wondering if there's a better way to go about getting the mappings for an index; I think we're essentially copying the mappings for a couple of tests. Perhaps a helper method that can get the mappings from the json resource files, and extract out just the `properties`?

Comment (eirsep) [reply src/test/java/org/opensearch/securityanalytics/resthandler/SecureThreatIntelMonitorRestApiIT.java:74]: removed this altogether

File: src/test/java/org/opensearch/securityanalytics/resthandler/SecureThreatIntelMonitorRestApiIT.java (hunk 1 above)
Comment (eirsep) [reply src/test/java/org/opensearch/securityanalytics/resthandler/SecureThreatIntelMonitorRestApiIT.java:55]: done

File: src/test/java/org/opensearch/securityanalytics/resthandler/SecureThreatIntelMonitorRestApiIT.java [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+import static org.opensearch.securityanalytics.resthandler.ThreatIntelMonitorRestApiIT.randomIocScanMonitorDto;
+
+public class SecureThreatIntelMonitorRestApiIT extends SecurityAnalyticsRestTestCase {
+    static String SECURITY_ANALYTICS_FULL_ACCESS_ROLE = "security_analytics_full_access";
+    static String SECURITY_ANALYTICS_READ_ACCESS_ROLE = "security_analytics_read_access";
+    static String TEST_HR_BACKEND_ROLE = "HR";
+
+    static String TEST_IT_BACKEND_ROLE = "IT";
+
+    static Map<String, String> roleToPermissionsMap = Map.ofEntries(
```
[the remark this answers is not in the document]
Comment (eirsep) [reply src/test/java/org/opensearch/securityanalytics/resthandler/SecureThreatIntelMonitorRestApiIT.java:49]: removed

**Review by jowg-amazon — APPROVED** [review]

**Review by AWSHurneyt — APPROVED** [review]

**Pull request merged**